embulk-jruby-strptime/src/main/java/org/embulk/spi/time/StrptimeParser.java

```diff
@@ -0,0 +1,883 @@
+package org.embulk.spi.time;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.util.EnumSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.embulk.spi.time.lexer.StrptimeLexer;
+
+/**
+ * This is Java implementation of ext/date/date_strptime.c in Ruby v2.3.1.
+ * @see https://github.com/ruby/ruby/blob/394fa89c67722d35bdda89f10c7de5c304a5efb1/ext/date/date_strptime.c
+ *
+ * TODO
+ * This class is tentatively required for {@code TimestampParser} class.
+ * After embulk-jruby-strptime will be merged into JRuby, embulk-jruby-strptime
+ * will be removed from Embulk.
```

Comment:
@dmikurube Sorry for the late response.. As talked directly about it to you, I understand your concerns.

I'm OK to keep it in Embulk. I agree with your thoughts. The reason why I tried to merge it into JRuby is because I wanted to reduce the maintenance cost for us. To do that, I'm also OK to accept the risk that you concerned. If Embulk core would use JRuby's and might be affected by the future updates, we would fix or fallback our original one at that time.

How do you think that we introduce an option to switch strptime implementation between our original and JRuby's. Users can specify it in system config or somewhere. The option can help the above falling back. The default value of the option is JRuby's.

How about it?